src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrResult = [];
+  const STATE_PREVIOS = { ...state };
+
+  for (const OBJ_ACTION of actions) {
+    if (OBJ_ACTION.type === 'addProperties') {
+      Object.assign(STATE_PREVIOS, OBJ_ACTION.extraData);
+      arrResult.push({ ...STATE_PREVIOS });
+    }
+
+    if (OBJ_ACTION.type === 'removeProperties') {
+      for (const key of OBJ_ACTION.keysToRemove) {
+        delete STATE_PREVIOS[key];
+      }
+      arrResult.push({ ...STATE_PREVIOS });
+    }
+
+    if (OBJ_ACTION.type === 'clear') {
+      for (const key in STATE_PREVIOS) {
+        delete STATE_PREVIOS[key];
+      }
+      arrResult.push({ ...STATE_PREVIOS });
```

Comment:
The pushing to `arrResult` is repeated in every case. This action should be performed once after the `switch` statement to adhere to the 'Don't Repeat Yourself' (DRY) principle.